code/game/machinery/computer/arcade/orion.dm

```diff
@@ -0,0 +1,524 @@
+		gamer.mind?.adjust_experience(/datum/skill/gaming, 50) // cheevos make u better
+
+		if(!isnull(GLOB.data_core.general))
+			for(var/datum/data/record/insanity_records in GLOB.data_core.general)
+				if(insanity_records.fields["name"] == gamer.name)
+					insanity_records.fields["m_stat"] = "*Unstable*"
+					return
+
+/obj/machinery/computer/arcade/orion_trail/ui_interact(mob/user, datum/tgui/ui)
+	. = ..()
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "OrionGame", name)
+		ui.open()
+
+/obj/machinery/computer/arcade/orion_trail/ui_assets(mob/user)
+	return list(
+		get_asset_datum(/datum/asset/spritesheet/moods),
+	)
+
+/obj/machinery/computer/arcade/orion_trail/ui_data(mob/user)
+	var/list/data = list()
+	data["gamestatus"] = gameStatus
+
+	data["engine"] = engine
+	data["turns"] = turns
+	data["hull"] = hull
+	data["electronics"] = electronics
+	data["food"] = food
+	data["fuel"] = fuel
+	data["lings_suspected"] = lings_suspected
+
+	data["eventname"] = event?.name
+	data["eventtext"] = event?.text
+	data["buttons"] = event?.event_responses
+
+	data["spaceport_raided"] = spaceport_raided
+
+	data["reason"] = reason
+
+	return data
+
+/obj/machinery/computer/arcade/orion_trail/ui_static_data(mob/user)
+	var/list/static_data = list()
+	static_data["gamename"] = name
+	static_data["emagged"] = obj_flags & EMAGGED
+	static_data["settlers"] = settlers
+	static_data["settlermoods"] = settlermoods
+	return static_data
+
+/obj/machinery/computer/arcade/orion_trail/ui_act(action, list/params)
+	. = ..()
+	if(.)
+		return
+	if(!iscarbon(usr))
+		return
+
+	. = TRUE
+
+	var/mob/living/carbon/gamer = usr
+
+	var/gamer_skill_level = 0
+	var/gamer_skill = 0
+	var/gamer_skill_rands = 0
+
+	if(gamer?.mind)
+		gamer_skill_level = gamer.mind.get_skill_level(/datum/skill/gaming)
+		gamer_skill = gamer.mind.get_skill_modifier(/datum/skill/gaming, SKILL_PROBS_MODIFIER)
+		gamer_skill_rands = gamer.mind.get_skill_modifier(/datum/skill/gaming, SKILL_RANDS_MODIFIER)
+
+	var/xp_gained = 0
+
+	if(event)
+		event.response(action)
+		if(!settlers.len || food <= 0 || fuel <= 0)
+			set_game_over(gamer)
+			return
+		new_settler_mood() //events shake people up a bit and can also change food
+		update_static_data(usr)
+		return TRUE
+	switch(action)
+		if("start_game")
+			if(gameStatus != ORION_STATUS_START)
+				return
+			newgame()
+		if("instructions")
+			if(gameStatus != ORION_STATUS_START)
+				return
+			gameStatus = ORION_STATUS_INSTRUCTIONS
+		if("back_to_menu") //back to the main menu
+			gameStatus = ORION_STATUS_START
+			if(gameStatus == ORION_STATUS_GAMEOVER)
+				event = null
+				reason = null
+				food = 80
+				fuel = 60
+				settlers = list("Harry","Larry","Bob")
+		if("continue")
+			if(turns >= ORION_TRAIL_WINTURN)
+				win(usr)
+				xp_gained += 34
+				return
+			usr?.mind?.adjust_experience(/datum/skill/gaming, xp_gained+1)
```

Comment:
use gamer here

Reply:
also stop abusing ?, you should only use that where it's actually sensible for the values to be potentially null.